aten/src/ATen/native/quantized/cpu/qlinear.cpp

```diff
@@ -1185,14 +1174,45 @@ static at::Tensor linear_int8_with_onednn_weight(
       at::native::itensor_view_from_dense(other.value().reshape({-1, other.value().size(dim - 1)})) :
       empty_tensor;
 
+  // Fast path with cache of params
+  static const char* env_var = std::getenv(CACHE_ONEDNN_CONTEXT_FLAG);
+  static const std::string cache_flag_str = env_var ? std::string(env_var) : "";
+  static const bool enable_primitive_cache = cache_flag_str != "" && cache_flag_str == "1";
+  static std::unordered_map<int64_t, QlinearForwardParams> qlinear_forward_params_map;
+  int64_t weight_addr = at::native::data_ptr_from_mkldnn(onednn_weight);
+  if (enable_primitive_cache && qlinear_forward_params_map.find(weight_addr) != qlinear_forward_params_map.end()) {
+    RECORD_FUNCTION("qlinear_fast_path", c10::ArrayRef<c10::IValue>({}));
+    auto& params = qlinear_forward_params_map[weight_addr];
+    auto& args = params.args;
+    args[DNNL_ARG_SRC] = std::move(src);
+    args[DNNL_ARG_DST] = std::move(dst);
+    if (binary_post_op == "add") {
+      args[DNNL_ARG_ATTR_MULTIPLE_POST_OP(0) | DNNL_ARG_SRC_1] = std::move(src1);
+    }
+    params.primitive.execute(ideep::stream::default_stream(), args);
+    return dim == 2 ? output : output.reshape(output_size);
```

Comment:
try to use `resize_()`, which returns the same `TensorImpl`

Reply:
Thanks. Updated.